rfcs/text/enzyme-to-rtl.md

```diff
@@ -0,0 +1,162 @@
+      "Enzyme test"
+    );
+  });
+});
+```
+
+The two libraries have slightly different APIs, but the tests are fairly similar and the effort required to convert them should be minimal.
+
+# Motivation
+
+The main motivation for us making this conversion is that, as previously mentioned, Enzyme does not yet support React 17. It is well over a year after it was released, and it does not look like support will be added in the near future. React 18 is in beta and we will be blocked from upgrading to that as well when it is released. It could be years before Enzyme add support for that, if ever.
+
+Some of our consumers are already using React 17 and we need to start adding support it for ASAP, as a widely used component library within Sage we can't afford to fall behind on basic things such as React support. A number of issues have already been raised by consumers about support for React 17 in some of our components.
+
+At Sage many of our modern React projects are now using React Testing Library, and it would be wise to use the same utilities across projects so developers across teams can support each other more easily. React Testing Library is also recommended by the architecture team for our React projects.
+
+Also using Jest with React Testing Library is now the [recommended setup](https://reactjs.org/docs/testing.html#tools) in the React documentation.
+
+[This section](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme#why-should-i-use-react-testing-library) in the React Testing Library docs also lists some great reasons to convert from Enzyme.
+
+# Detailed design
+
+In order to make the transition as smooth as possible, we should convert our component tests incrementally, and therefore need to temporarily support both testing utilities.
+
+## Config setup
+
+To add in React Testing Library, we just need to install `@testing-library/react` and `@testing-library/jest-dom` as dev depencencies and no further setup is needed to get started. We can simply import the modules we need into our test files, such as:
+
+```javascript
+import { render, screen } from "@testing-library/react";
+```
+
+No modifications would be needed to our Jest config either, until we finish the conversion and remove anything related to Enzyme.
+
+We currently wrap components in theme or i18n providers on a test-by-test basis, but if we decide we want to wrap all components in these by default, we can create a custom render method as desribed [here](https://testing-library.com/docs/react-testing-library/setup/#custom-render) in the React Testing Library docs.
+
+If we were to do so, we would likely want to wrap our components in the `CarbonProvider` component (to provide the theme and the Design Token CSS variables) and the `I18nProvider` component for translations, like this:
+
+```jsx
+const CustomProviders = ({ children }) => {
+  return (
+    <CarbonProvider theme="mint">
```

Comment:
I know this is just a demo of a potential implementation but I think it would be good to have `theme` and `locale` as props of `CustomProvider`. `mint` and `enGB` are already the default values for the `CarbonProvider` and `I18nProvider` but it might be more explicit also have default values of `"mint"` and `enGB` if you do choose to add the props to the top level destructure

Reply:
Thanks @edleeks87 - it's a good idea in theory but unfortunately using the `wrapper` option with the RTL `render` function, it only passes `children` as a prop, you can't pass any other props in. You can see that [here](https://github.com/testing-library/react-testing-library/blob/8f17a2bc4fa9c2c17a623b38ded9ab18b086e457/src/pure.js#L56).

There's really no need to change the theme in the unit tests anyway, they should be for testing the functionality of the code. Chromatic will take care of testing the differences between themes for us. 

Translations wise, if we need to use a different locale for a certain test, we can simply just not use this custom render function, just use the normal `render` from RTL and an `I18nProvider` with whatever locale we need. 

Good point about the defaults but I just added them specifically in for the sake of the example here, in reality we don't have to have theme if we do just want to go with the default values 👍🏻

Reply:
Ah fair point I hadn't realised RTL's render constrained it like that, I'm happy to leave as is then 👍